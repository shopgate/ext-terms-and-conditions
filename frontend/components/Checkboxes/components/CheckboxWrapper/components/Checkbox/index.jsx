import React from 'react';
import PropTypes from 'prop-types';
import SharedCheckbox from '@shopgate/pwa-ui-shared/Checkbox';
import styles from './style';

/**
 * CheckBox component
 * @param {Object} props props
 * @returns {JSX}
 */
const Checkbox = props => (
  <div className={styles.wrapper}>
    <label
      onClick={() => props.onChange(props.wrapperIndex, props.value)}
      className={styles.label}
      style={{ color: `${props.textColor}` }}
    >
      <SharedCheckbox
        className={styles.box}
        type="checkbox"
        value={props.value}
        checked={props.checked}
      />
      <span className={styles.labelSpan}>{props.label}</span>
    </label>
  </div>
);

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
  textColor: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  wrapperIndex: PropTypes.number,
};

Checkbox.defaultProps = {
  checked: false,
  onChange: () => {},
  wrapperIndex: null,
};

export default Checkbox;
