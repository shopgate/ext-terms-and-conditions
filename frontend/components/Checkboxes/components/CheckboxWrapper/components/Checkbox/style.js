import { css } from 'glamor';

const wrapper = css({
  paddingTop: '5px',
}).toString();

const label = css({
  display: 'inline-block',
  fontFamily: 'inherit',
  fontSize: '16px',
  lineHeight: '160%',
}).toString();

const labelSpan = css({
  display: 'block',
  overflow: 'auto',
}).toString();

const box = css({
  float: 'left',
  paddingRight: '8px',
}).toString();

export default {
  box,
  label,
  labelSpan,
  wrapper,
};
