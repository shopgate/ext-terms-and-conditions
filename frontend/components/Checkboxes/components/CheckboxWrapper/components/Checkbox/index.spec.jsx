import React from 'react';
import { mount } from 'enzyme';
import { createMockStore } from '@shopgate/pwa-common/store';
import { Provider } from 'react-redux';
import Checkbox from './index';

const store = createMockStore();

describe('Checkbox', () => {
  it('should render', () => {
    const component = mount((
      <Provider store={store}>
        <Checkbox
          label="mock"
          textColor="#000"
          value={0}
        />
      </Provider>
    ));
    expect(component.html()).not.toBe(null);
    expect(component).toMatchSnapshot();
  });
  it('should be clickable', () => {
    const mockCallBack = jest.fn();
    const component = mount((
      <Provider store={store}>
        <Checkbox
          label="mock"
          onChange={mockCallBack}
          textColor="#000"
          value={0}
        />
      </Provider>
    ));
    expect(component.html()).not.toBe(null);
    component.find('label').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
    expect(component).toMatchSnapshot();
  });
});
