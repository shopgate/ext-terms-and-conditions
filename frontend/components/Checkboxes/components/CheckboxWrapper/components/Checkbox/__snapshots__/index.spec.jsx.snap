// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should be clickable 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Checkbox
    checked={false}
    label="mock"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            null,
            0,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    textColor="#000"
    value={0}
    wrapperIndex={null}
  >
    <div
      className="css-1ye9cz5"
    >
      <label
        className="css-b2ryqc"
        onClick={[Function]}
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <Checkbox
          checked={false}
          checkedClassName="css-5oj2eb"
          className="css-1pwqlfb"
          type="checkbox"
          unCheckedClassName="css-1n9afhg"
          value={0}
        >
          <Checkbox
            checked={false}
            checkedIcon={
              <Checked
                className="css-37nc4w css-5oj2eb"
              />
            }
            className="css-1pwqlfb"
            disabled={false}
            label={null}
            labelPosition="left"
            onCheck={[Function]}
            type="checkbox"
            uncheckedIcon={
              <Unchecked
                className="css-37nc4w css-1n9afhg"
              />
            }
            value={0}
          >
            <div
              aria-hidden={true}
              className="css-1pwqlfb"
              onClick={[Function]}
            >
              <Unchecked
                className="css-37nc4w css-1n9afhg"
              >
                <Icon
                  className="css-37nc4w css-1n9afhg"
                  color={null}
                  content="<path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"/>"
                  size="inherit"
                  viewBox="0 0 24 24"
                >
                  <svg
                    className="css-yys9hb css-37nc4w css-1n9afhg"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"/>",
                      }
                    }
                    style={
                      Object {
                        "fill": null,
                        "fontSize": "inherit",
                      }
                    }
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </Icon>
              </Unchecked>
            </div>
          </Checkbox>
        </Checkbox>
        <span
          className="css-64ph5z"
        >
          mock
        </span>
      </label>
    </div>
  </Checkbox>
</Provider>
`;

exports[`Checkbox should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Checkbox
    checked={false}
    label="mock"
    onChange={[Function]}
    textColor="#000"
    value={0}
    wrapperIndex={null}
  >
    <div
      className="css-1ye9cz5"
    >
      <label
        className="css-b2ryqc"
        onClick={[Function]}
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <Checkbox
          checked={false}
          checkedClassName="css-5oj2eb"
          className="css-1pwqlfb"
          type="checkbox"
          unCheckedClassName="css-1n9afhg"
          value={0}
        >
          <Checkbox
            checked={false}
            checkedIcon={
              <Checked
                className="css-37nc4w css-5oj2eb"
              />
            }
            className="css-1pwqlfb"
            disabled={false}
            label={null}
            labelPosition="left"
            onCheck={[Function]}
            type="checkbox"
            uncheckedIcon={
              <Unchecked
                className="css-37nc4w css-1n9afhg"
              />
            }
            value={0}
          >
            <div
              aria-hidden={true}
              className="css-1pwqlfb"
              onClick={[Function]}
            >
              <Unchecked
                className="css-37nc4w css-1n9afhg"
              >
                <Icon
                  className="css-37nc4w css-1n9afhg"
                  color={null}
                  content="<path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"/>"
                  size="inherit"
                  viewBox="0 0 24 24"
                >
                  <svg
                    className="css-yys9hb css-37nc4w css-1n9afhg"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"/>",
                      }
                    }
                    style={
                      Object {
                        "fill": null,
                        "fontSize": "inherit",
                      }
                    }
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </Icon>
              </Unchecked>
            </div>
          </Checkbox>
        </Checkbox>
        <span
          className="css-64ph5z"
        >
          mock
        </span>
      </label>
    </div>
  </Checkbox>
</Provider>
`;
